src/plugins/intel_gpu/src/graph/impls/onednn/reduction_onednn.cpp

```diff
@@ -27,8 +27,18 @@ struct reduction_onednn : typed_primitive_onednn_impl<reduce, dnnl::reduction::d
     static std::shared_ptr<dnnl::reduction::desc> get_reduction_descriptor(const reduce_node& arg) {
         auto prim = arg.get_primitive();
         auto& input = arg.get_dependency(0);
-        auto input_md = onednn::layout_to_memory_desc(input.get_output_layout());
-        auto output_md = onednn::layout_to_memory_desc(arg.get_output_layout());
+        auto input_layout = input.get_output_layout();
+        auto output_layout = arg.get_output_layout();
+
+        for (size_t idx = 0 ; idx < prim->axes.size(); idx++) {
+            if (output_layout.get_dim(prim->axes[idx]) != 1) {
+                reorder_unreduced_axis_no_fusion(input_layout, output_layout, prim->axes);
+                break;
+            }
+        }
```

Comment:
* Do we need a for loop here?
* I think adding some assert would be better to understand
* The function name `reorder_unreduced_axis_no_fusion` is difficult to understand. Can you add some comment?

Reply:
The loop is to check all axes whether the output size is matching to input size by reduced axes.
comments are added.